src/test/java/io/jenkins/plugins/coverage/CoverageChecksPublisherTest.java

```diff
@@ -0,0 +1,142 @@
+package io.jenkins.plugins.coverage;
+
+import io.jenkins.plugins.checks.api.ChecksConclusion;
+import io.jenkins.plugins.checks.api.ChecksDetails;
+import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+import io.jenkins.plugins.coverage.targets.*;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CoverageChecksPublisherTest {
+    @Test
+    public void shouldConstructChecksDetailsWithLineAndMethodCoverage() {
+        CoverageResult result = createCoverageResult((float)0.6, (float)0.4);
+        when(result.getPreviousResult()).thenReturn(null);
+
+        CoverageAction action = mock(CoverageAction.class);
+        when(action.getResult()).thenReturn(result);
+        when(action.getAbsoluteUrl()).thenReturn("http://127.0.0.1:8080/job/pipeline-coding-style/job/PR-3/49/coverage");
+
+        ChecksDetails expectedDetails = new ChecksDetailsBuilder()
+                .withName("Code Coverage")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withDetailsURL("http://127.0.0.1:8080/job/pipeline-coding-style/job/PR-3/49/coverage")
+                .withOutput(new ChecksOutputBuilder()
+                        .withTitle("Line coverage of 60%.")
+                        .withSummary("")
+                        .withText("## Conditional\n* :white_check_mark: Coverage: 40%\n"
```

Comment:
(*Conditional* actually is the wrong name, but this is something that needs to be changed in the coverage API and not in your PR)